packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -235,10 +239,18 @@ export class ProjectService {
             return { warehouseClient: existingClient, sshTunnel };
         }
         // otherwise create a new client and cache for future use
+        const credentialsWithWarehouse =
+            credentials.type === WarehouseTypes.SNOWFLAKE
+                ? {
+                      ...warehouseSshCredentials,
+                      warehouse:
+                          snowflakeVirtualWarehouse || credentials.warehouse,
+                  }
+                : warehouseSshCredentials;
```

Comment:
This is the fix , https://github.com/lightdash/lightdash/pull/8255/files#r1427337882